### PR TITLE
Resolve #1186 -- Fixed HasEmptyScript Crash

### LINKED ISSUE #1186: Spell Script Crash
The Spell class incorrectly handles the assigning of spell Scripts to their empty versions, leading to a crash when a spell, such as a passive, has their loading skipped.

To solve this, these changes are recommended:

```public ISpellScript Script { get; private set; } = new SpellScriptEmpty();
public bool HasEmptyScript { get; private set; }

public Spell(Game game, IObjAiBase owner, string spellName, byte slot)
        {
           ...
           if (CastInfo.SpellSlot != (int)SpellSlotType.PassiveSpellSlot)
           {
              //Set the game script for the spell
              LoadScript();
           }
           HasEmptyScript = Script.GetType() == typeof(SpellScriptEmpty);
        }
        
        public void LoadScript()
        {
           ...
           Script = _scriptEngine.CreateObject<ISpellScript>("Spells", SpellName) ?? Script;
```

### PR DESCRIPTION
**Pull request by cabeca1143**
Fixed HasEmptyScript crash when loading CharScripts due to the "Script" variable being null.

Resolve #1186

### CODE REVIEW

File: GameServerCore/Domain/GameObjects/Spell/ISpell.cs [hunk 1]
```diff
@@ -52,7 +52,7 @@ public interface ISpell: IUpdate
         /// <summary>
         /// Whether or not the script for this spell is the default empty script.
         /// </summary>
-        bool HasEmptyScript { get; }
+        bool HasEmptyScript { get; set; }
```
Comment (danil179) [inline GameServerCore/Domain/GameObjects/Spell/ISpell.cs]: You shouldn't expose this property to be public.

**Review by danil179 — CHANGES_REQUESTED** [review]
There should be another solution.

**Comment by lzardy** [comment]
Related #1186
Edit the original post to include `Resolve #1186`

File: GameServerCore/Domain/GameObjects/Spell/ISpell.cs (hunk 1 above)
Comment (cabeca1143) [reply GameServerCore/Domain/GameObjects/Spell/ISpell.cs]: Resolved

File: GameServerLib/GameObjects/Spell/Spell.cs [hunk 2]
```diff
@@ -108,6 +114,8 @@ public void LoadScript()
 
             Script = _scriptEngine.CreateObject<ISpellScript>("Spells", SpellName) ?? new SpellScriptEmpty();
 
+            HasEmptyScript = Script.GetType() == typeof(SpellScriptEmpty);
```
Comment (danil179) [inline GameServerLib/GameObjects/Spell/Spell.cs:117]: This can be removed later.

**Review by danil179 — APPROVED** [review]
Approved. I'm not exactly sure why the CharScript is in ObjAiBase and I'm sure that many incompatibilities will be discovered (as Script is null or so in the Spell intself).

**Pull request merged**